node/src/actors/node.rs

```diff
@@ -12,12 +12,17 @@ use crate::{
     config_mngr, signature_mngr, storage_mngr,
 };
 use witnet_config::config::Config;
+use witnet_validations::witnessing::validate_witnessing_config;
 
 /// Function to run the main system
 pub fn run(config: Arc<Config>, callback: fn()) -> Result<(), failure::Error> {
     // Init system
     let system = System::new();
 
+    // Perform some initial validations on the configuration
+    let witnessing_config = config.witnessing.clone().into_config();
+    validate_witnessing_config(&witnessing_config)?;
```

Comment:
This results in the following logs when starting a node with an invalid proxy url:

```
[2022-08-12T10:14:51Z DEBUG witnet_data_structures] Set environment to development
[2022-08-12T10:14:51Z INFO  witnet_config::config] The default unproxied HTTP transport for retrieval is enabled.
[2022-08-12T10:14:51Z INFO  witnet_config::config] Configuring retrieval proxies: ["http://183.239.61.204:9091", "http://183.239.61.204:90911"]
[2022-08-12T10:14:51Z INFO  witnet_config::config] Paranoid retrieval percentage is set to 51%
Error: The following transport addresses are invalid:
- http://183.239.61.204:90911 (invalid port number)
```

The error only mentions "transport addresses", it doesn't mention that this is related to proxies or witnessing. So doing something like this would add a bit of context:

```suggestion
    validate_witnessing_config(&witnessing_config).map_err(|e| {
        failure::format_err!("Error when validating witnessing config: {}", e)
    })?;
```

Which results in this error message:

```
Error: Error when validating witnessing config: The following transport addresses are invalid:
- http://183.239.61.204:90911 (invalid port number)
```

Reply:
I have directly changed the `impl Display for WitnessingConfigError` instead. It was totally missing a first mention to the error happening while validating witnessing configuration. This will solve the thing for both the node and the wallet.